rclcpp/src/rclcpp/duration.cpp

```diff
@@ -182,6 +183,34 @@ Duration::operator-(const rclcpp::Duration & rhs) const
     rcl_duration_.nanoseconds - rhs.rcl_duration_.nanoseconds);
 }
 
+void
+bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
+{
+  auto abs_dns = static_cast<uint64_t>(std::abs(dns));
+  auto abs_scale = std::abs(scale);
+
+  if (abs_scale > 1.0 && abs_dns > static_cast<uint64_t>(max / abs_scale)) {
+    if ((dns > 0 && scale > 0) || (dns < 0 && scale < 0)) {
+      throw std::overflow_error("duration scaling leads to int64_t overflow");
+    } else {
+      throw std::underflow_error("duration scaling leads to int64_t underflow");
+    }
+  }
+}
+
+Duration
+Duration::operator*(double scale) const
+{
+  if (!std::isnormal(scale)) {
```

Comment:
@tfoote is it appropriate to prevent users from having a scale of 0 here? Seems too strict

Reply:
Yeah, a zero scale value would be valid. I think `isfinite` would be a better check.

Reply:
I agree, in theory zero is valid to scale although this operation is meaningless. Ok, I'll update code from isnormal to isfinite which allow both normal and zero.